### PR TITLE
Fix: Announce only the first error message for better accessibility

### PR DESCRIPTION
**Pull request by azizChebbi**
## Issue
This pull request addresses an accessibility issue where all form error messages were concatenated and announced simultaneously, overwhelming users relying on screen readers like VoiceOver and TalkBack. 
This is the issue link: https://github.com/flutter/flutter/issues/156340

## Update
The change ensures that only the first error message is announced, providing a more manageable and user-friendly experience.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/form.dart [hunk 1]
```diff
@@ -367,7 +367,9 @@ class FormState extends State<Form> {
       if (!validateOnFocusChange || !field._focusNode.hasFocus) {
         final bool isFieldValid = field.validate();
         hasError = !isFieldValid || hasError;
-        errorMessage += field.errorText ?? '';
+        if (errorMessage.isEmpty) {
+          errorMessage = field.errorText ?? '';
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/form.dart:371]: I don't have access to a real device now, but we should check for native behavior. If the concatenation string was confusing, maybe we should concate with 2 line break in between each message

Comment (azizChebbi) [reply packages/flutter/lib/src/widgets/form.dart:371]: but even with adding the 2 line break, we are still facing the same issue, that the voice over is revealing all the error messages at once, we don't actually want that behavior.
We reveal only the first error, after that the user fills the field and pass to the next one.

Comment (azizChebbi) [reply packages/flutter/lib/src/widgets/form.dart:371]: @chunhtai what do you think ?
This change is really crucial as we're working with a big institute that provides benefits for the blind and visually impaired, and they suggested this accessibility improvement. It's key that our system works well for them. 
And thanks in advance!

**Comment by hannah-hyj** [comment]
I'm trying to check the native behavior. The closest scenario I could find is adding a credit card to the Play Store/App Store.  https://github.com/user-attachments/assets/b474e401-d52e-49b5-b233-149fe6aca0c4

Android only announces **one** error message for the text field you're editing.   Also, the error messages I tested in this case are live when editing. In the flutter TextFormField widget, you can set the error messages to be live when editing or appear simultaneously when finally submitting the form. 

I agree that we should only announce the error message from the first error form field so it's not too overwhelming.

File: packages/flutter/lib/src/widgets/form.dart [hunk 2]
```diff
@@ -367,7 +367,9 @@ class FormState extends State<Form> {
       if (!validateOnFocusChange || !field._focusNode.hasFocus) {
         final bool isFieldValid = field.validate();
         hasError = !isFieldValid || hasError;
-        errorMessage += field.errorText ?? '';
+        if (errorMessage.isEmpty) {
```
Comment (hannah-hyj) [inline packages/flutter/lib/src/widgets/form.dart:370]: You can add a comment here for this change.

**Comment by hannah-hyj** [comment]
Please also add a test in [form_test.dart](https://github.com/flutter/flutter/pull/123373/files#diff-0a50dbd2cdbd6a623a9b778f2df2a778d7637f22bf080e23cfbb52155929c386)

File: packages/flutter/lib/src/widgets/form.dart (hunk 2 above)
Comment (azizChebbi) [reply packages/flutter/lib/src/widgets/form.dart:370]: done!

**Comment by azizChebbi** [comment]
> Please also add a test in form_test.dart

done! cc @hannah-hyj

File: packages/flutter/lib/src/widgets/form.dart (hunk 2 above)
Comment (hannah-hyj) [reply packages/flutter/lib/src/widgets/form.dart:370]: nit:  The comment `// Ensure that only the first error message gets announced to the user.` should be one line before 
`if (errorMessage.isEmpty) {`

**Comment by hannah-hyj** [comment]
Thank you for adding the test

I think this looks good to me except for a small comment.

**Comment by melWiss** [comment]
Hey, what's the status here? The pr seems okay to me, can we get this ready for the next release? our team is blocked and needs this one, cc @chunhtai @hannah-hyj

**Review by chunhtai — COMMENTED** [review]
I think this change make sense, but do you know what is the native form does in this case?

File: packages/flutter/lib/src/widgets/form.dart (hunk 2 above)
Comment (hannah-hyj) [reply packages/flutter/lib/src/widgets/form.dart:370]: This is not updated.

**Comment by hannah-hyj** [comment]
LGTM except for a nit. 


@chunhtai  
what is the native form does in this case?>>
https://github.com/flutter/flutter/pull/156399#issuecomment-2415321098  For this question I tested some native behavior but I didn't find an exact native equivalent for the flutter form widget to validate multiple form field at the same time when submitting the form.

**Review by hannah-hyj — CHANGES_REQUESTED** [review]
LGTM except one nit

**Review by hannah-hyj — APPROVED** [review]

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**